3ds Max/Max2Babylon/MaterialScripts.cs

```diff
@@ -20,41 +20,28 @@ public static class MaterialScripts
           l = custAttributes.count mat; 
           for i = 1 to l+1 do ( 
             if(mat.custAttributes[i] != undefined) then (
-                if(mat.custAttributes[i].name == attName) then return true
-            ) ) return false )";
+                if(mat.custAttributes[i].name == attName) then
+                ( 
+                    return true
+                )
+            ) )  return false )";
 
-        /// <summary>
-        /// Note : We may run a check on all Slate view, because race condition into script execution 
-        /// do not ensure the link beetwen material and bump are correctly set when material is updated.
-        /// </summary>
-        public static string CheckBumpMap = $@"fn checkBumpMap = (
-			local babylonAttName = ""Babylon Attributes""
-			for v=1 to sme.GetNumViews() do 
-			(
-				viewNode = sme.GetView(v)
-				for i=1 to viewNode.GetNumNodes() do
-				(
-					n = viewNode.GetNode i;
-					if(classof n.reference == Normal_Bump ) then
-				    (
-					   nbump = n.reference
-					   if(hasBabylonCustomAttribute nbump babylonAttName == false ) then
-					   (
-                            {BumpMapCAtDef}
-							custAttributes.add nbump BUMP_MAP_CAT_DEF
-					   )
-					) 
-				)	
-			)        
-        )";
+        public static string CheckBumpMap = $@"fn checkBumpMap map = (
+            local babylonAttName = ""Babylon Attributes""
+            if(map != undefined) then 
+            ( 
+              if( classof map == Normal_Bump ) then
+        	  (
+        		if( hasBabylonCustomAttribute map babylonAttName == false) then
+        		(
+                    {BumpMapCAtDef}
+        			custAttributes.add map BUMP_MAP_CAT_DEF
+        		))))";
 
-        /// There is NO WAY to get an even on BumpMap creation into the Slate editor. Only Material creation is triggered.
-        /// This is the reason we listen for parameters Material change
         private static string AddPhysicalBabylonUI =
         $@"if ( hasBabylonCustomAttribute maxMaterial ""Babylon Attributes"" == false) then ( 
-              {PhysicalBabylonCAtDef}
+             {PhysicalBabylonCAtDef}
               custAttributes.add maxMaterial PHYSICAL_MATERIAL_CAT_DEF 
-              when parameters maxMaterial change do checkBumpMap()
```

Comment:
I'm assuming that this is what's causing our stackoverflow? We change the material when we execute our script checkBumpMap()? I would hope that there's some other workaround than making this behavior explicit... i.e this makes a workflow where one is programmatically creating materials a bit more cumbersome, as now one has to make an additional step to add the material custom attributes as well...

If it can't be helped, then hey, its a recurring theme that specifying material attributes outside the capabilities of the DCC's material model is not fun, regardless of the DCC tool it seems...

Whatever solution you choose to proceed with should work